src/repr/src/explain_new/mod.rs

```diff
@@ -657,17 +664,29 @@ where
     let mut first_rows = Vec::with_capacity(20);
     for _ in 0..20 {
         if let Some((row, diff)) = rows.next() {
-            row_count += diff;
+            row_count += diff.abs();
             first_rows.push((row, diff));
         }
     }
-    let rest_of_row_count = rows.into_iter().map(|(_, diff)| diff).sum::<crate::Diff>();
-    if rest_of_row_count != 0 {
-        writeln!(f, "{}total_rows: {}", ctx, row_count + rest_of_row_count)?;
-        writeln!(f, "{}first_rows:", ctx)?;
-        ctx.indented(move |ctx| write_first_rows(f, &first_rows, ctx))?;
+    if row_count > 0 {
```

Comment:
All other parts of the `EXPLAIN AS TEXT` (including the current version of `fmt_text_constant_rows`) follow the `$Op $attr=$val` output convention for one-liner, and
```text
$Op
  $attr: $val
  ...
```
convention if at least one of the attribute values has structure that requires multi-line output (structs, long lists as in the current `Constant` output). We should probably stick with that for consistency. If you only want to indicate that 
```
Constant
```
represents the empty collection, I suggest grepping for `"{}Constant` and adding a variant that emits
```
Constant ∅
```
if `rows` is empty, and leave `fmt_text_constant_rows` unchanged.

Reply:
Ok, I moved the empty collection handling logic out of `fmt_text_constant_rows`.

I'm not sure how many people know the `∅` sign, so I would leave it `<empty>`.